stdlib/REPL/src/docview.jl

```diff
@@ -108,7 +108,7 @@ function Markdown.term(io::IO, msg::Message, columns)
     printstyled(io, msg.msg; msg.fmt...)
 end
 
-function trimdocs(md::Markdown.MD, brief::Bool)
+function trimdocs(md, brief::Bool)
     brief || return md
     md, trimmed = _trimdocs(md, brief)
     if trimmed
```

Comment:
Since `md` is not a `Markdown.MD` anymore, it's a bit misleading to name it `md`.  Also, `if trimmed` block expect `md` to be a `Markdown.MD`.  You wouldn't hit that block since `trimmed` would be always `false` for non-`Markdown.MD` object.  But it is not immediately obvious.  So, somebody might try to refactor this to the one before the patch.  I think a cleaner approach would be to add

```julia
trimdocs(doc, brief::Bool) = doc
```

instead.

Reply:
I actually had something similar on my first pass, but changed it to be the absolute minimal change (in the hope that simple is easier to pass muster). I'd be happy to change more, though, if that's preferred.

Reply:
Minimal does not mean maintainable.  The current patch has the problems I mentioned.  I think adding an extra method is cleaner.

Reply:
Sure — that's why I pinged Tim to ask about original intent. The fallback `_trimdocs` was obviously meant to serve that purpose, so I can just change that to be a level higher in the call chain.

Commit incoming.

Reply:
> The fallback `_trimdocs` was obviously meant to serve that purpose

I'm not sure. It'd imagine that Tim would have tested it if so. It could just be an implementation detail for handling recursion.

Reply:
You're right. Following comment was relevant. I'd assumed all Markdown-parsed content was an `MD` type, but there are more types.